helix-term/src/ui/prompt.rs

```diff
@@ -450,6 +453,25 @@ impl Component for Prompt {
             compositor.pop();
         })));
 
+        if self.pending_register {
+            // C-r was pressed previously; this keystroke is the register
+            self.pending_register = false;
+            cx.editor.autoinfo = None;
+
+            if let KeyEvent {
+                code: KeyCode::Char(register),
+                modifiers: KeyModifiers::NONE,
+            } = event.into()
+            {
+                if let Some(text) = cx.editor.registers.get(register) {
+                    self.insert_str(&text.read()[0]);
+                }
+            }
+
+            (self.callback_fn)(cx, &self.line, PromptEvent::Update);
+            return EventResult::Consumed(None);
+        }
```

Comment:
This feels weird to me and if more and more states are being added it might be unmaintainable, I think it might be better to have some sort of state machine (like using `enum`) here to make sure we can't go wrong. But maybe someone else might have a better solution for this.